command/server/config.go

```diff
@@ -97,6 +103,23 @@ func (c *Config) Merge(c2 *Config) *Config {
 		result.Telemetry = c2.Telemetry
 	}
 
+	// merging this boolean via an OR operation
+	result.DisableMlock = c.DisableMlock
+	if c2.DisableMlock {
+		result.DisableMlock = c2.DisableMlock
+	}
+
+	// merge these integers via a MAX operation
```

Comment:
The configuration value should take precedent if specified. Otherwise its impossible to set a value less than the default.